dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/HibernateTrackedEntityStore.java

```diff
@@ -1021,15 +1037,26 @@ private String getFromSubQueryLimitAndOffset(PageParams pageParams) {
           .append(pageParams.getOffset())
           .append(SPACE)
           .toString();
-    } else if (systemMaxPageSize > 0) {
+    } else if (systemMaxLimit > 0) {
       return limitOffset
           .append(LIMIT)
           .append(SPACE)
-          .append(systemMaxPageSize)
+          .append(systemMaxLimit)
           .append(SPACE)
           .toString();
     }
 
     return limitOffset.toString();
   }
+
+  /** Returns the maximum TE retrieval limit. 0 no limit. */
```

Comment:
Maybe add that "the assumption is that only one of TET/Program will ever be set" so we don't forget 😅